.github/workflows/gem-build-and-install.yml

```diff
@@ -0,0 +1,86 @@
+name: Test gem build and installation
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  native-package:
+    name: Compile native gem
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        platform:
+          - x86_64-linux
+          # - arm-linux # needs https://github.com/matsadler/magnus/issues/17 ", "aarch64-linux", "x86_64-darwin", "arm64-darwin", "x64-mingw32", "x64-mingw-ucrt"
+          # - "aarch64-linux"
+
+          - "x86_64-darwin"
+          - "arm64-darwin"
+
+          # - "x64-mingw-ucrt"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
```

Comment:
You could use [`cargo-extra-cache` config as well](https://github.com/oxidize-rb/actions/blob/21b8acf931df3286b9cc4d3a7366b744f9bb4791/setup-ruby-and-rust/action.yml#L50), which should handle busting the cache as reliably as possible

Reply:
Thanks!